lte/cloud/go/services/lte/servicers/builder_servicer.go

```diff
@@ -493,3 +507,36 @@ func shouldEnableDNSCaching(dns *lte_models.GatewayDNSConfigs) bool {
 	}
 	return swag.BoolValue(dns.EnableCaching)
 }
+
+// getFederatedNetworkConfig returns the FEG configuration for this specific network.
+// If the network is not federated returns nil, but not error
+func getFederatedNetworkConfig(agNwID string, federatedId lte_models.FegNetworkID) (*feg_models.NetworkFederationConfigs, error) {
+	fedId := string(federatedId)
+	if federatedId == "" {
+		return nil, nil
+	}
+	fegCfg, err := configurator.LoadNetworkConfig(fedId, feg.FegNetworkType, lte_serdes.Network)
+	if err != nil || fegCfg == nil {
+		return nil, status.Errorf(
+			codes.Internal, "unable to retrieve config for federation network: %s", fedId)
+	}
+
+	networkFegConfigs, ok := fegCfg.(*feg_models.NetworkFederationConfigs)
+	if !ok || networkFegConfigs == nil {
+		return nil, status.Errorf(
+			codes.Internal, "invalid federation network config found for network: %s", fedId)
+	}
+	return networkFegConfigs, nil
+}
+
+// getNHroutesFromFeg extracts NH routes from the model into mconfig object
+func getNHroutesFromFeg(agNwID string, networkFegConfigs *feg_models.NetworkFederationConfigs) *lte_mconfig.NHRoutes {
+	if networkFegConfigs != nil {
+		for _, network := range networkFegConfigs.ServedNetworkIds {
+			if agNwID == network {
+				return &lte_mconfig.NHRoutes{PlmnMap: networkFegConfigs.NhRoutes}
+			}
+		}
+	}
+	return &lte_mconfig.NHRoutes{}
+}
```

Comment:
the two directional AGW Network <-> FeG Network config match check is a security/contract enforcement feature, you may not need it here, the check will be done anyway in real time & your configuration can just reflect AGW view and doesn't need to complicate itself with the check.